controllers/cruisecontroltask_controller.go

```diff
@@ -302,6 +348,69 @@ func (r *CruiseControlTaskReconciler) createCCOperation(
 	}, nil
 }
 
+// brokersJBODSelector filters out the JBOD and not JBOD brokers from a broker list based on the capacityConfig
+func brokersJBODSelector(brokerIDs []string, capacityConfigJSON string) (brokersJBOD []string, brokersNotJBOD []string, err error) {
+	// JBOD is generated by default
+	if capacityConfigJSON == "" {
+		return brokerIDs, nil, nil
+	}
+	brokerIsJBOD := make(map[string]bool)
```

Comment:
**Note**: I think we could have spared this map by directly appending to one of the returned slices and on -1 case return the `brokerIDs` slice as either the first or second return value depending on the case, but I don't force this optimization if we feel the current solution improves clarity.

Reply:
When I first made this function I thought that CC can do disk rebalance for the JBOD brokers. That is why I made this selector and I created different CCOperation for JBOD and not JBOD brokers. Unfortunately, later I found that CC does not capable to do disk rebalance when there is one not JBOD broker in the KafkaCluster because this rebalance endpoint is designed in another way. Basically, it would be enough to do a return false when we found a not JBOD broker. I kept that function in the first form because who knows the future maybe this will be changed on the CC side.

Reply:
Yeah, it is perfectly fine to return a more detailed answer with the broker break down.
I just wanted to highlight using the slices would be enough, the map is only a convenience/clarity thing at most, but it is fine for me anyway.